### PR TITLE
test: add e2e test "dynamically_provisioned_collocated_pod_test"

### LINKED ISSUE #91: Enable more e2e tests
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

Currently there is only one e2e test on https://github.com/kubernetes-sigs/azurefile-csi-driver/tree/master/test/e2e, need to write more e2e tests into https://github.com/kubernetes-sigs/azurefile-csi-driver/tree/master/test/e2e/testsuites

**Describe alternatives you've considered**

**Additional context**

/assign @ZeroMagic

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
add e2e test "dynamically_provisioned_collocated_pod_test"

**Which issue(s) this PR fixes**:

Fixes #91 

**Special notes for your reviewer**:

**Release note**:
```
add e2e test "dynamically_provisioned_collocated_pod_test"
```

### CODE REVIEW

**Comment by ZeroMagic** [comment]
```
• [SLOW TEST:184.278 seconds]
Dynamic Provisioning
/root/go/src/github.com/kubernetes-sigs/azurefile-csi-driver/test/e2e/dynamic_provisioning.go:30
  should create multiple PV objects, bind to PVCs and attach all to different pods on the same node
  /root/go/src/github.com/kubernetes-sigs/azurefile-csi-driver/test/e2e/dynamic_provisioning.go:6
------------------------------
• [SLOW TEST:46.705 seconds]
Dynamic Provisioning
/root/go/src/github.com/kubernetes-sigs/azurefile-csi-driver/test/e2e/dynamic_provisioning.go:30
  should create a volume on demand
  /root/go/src/github.com/kubernetes-sigs/azurefile-csi-driver/test/e2e/dynamic_provisioning.go:45
------------------------------

Ran 2 of 2 Specs in 230.983 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS
```

File: test/e2e/dynamic_provisioning.go [hunk 1]
```diff
@@ -63,4 +63,41 @@ var _ = Describe("Dynamic Provisioning", func() {
...
 		test.Run(cs, ns)
 	})
+
+	It("should create multiple PV objects, bind to PVCs and attach all to different pods on the same node", func() {
+		pods := []testsuites.PodDetails{
+			{
+				Cmd: "while true; do echo $(date -u) >> /mnt/test-1/data; sleep 1; done",
+				Volumes: []testsuites.VolumeDetails{
+					{
+						FSType:    "ext3",
```
Comment (andyzhangx) [inline test/e2e/dynamic_provisioning.go]: actually specify `FSType` does not make sense, we may remove this parameter and only run on case here?

Comment (andyzhangx) [reply test/e2e/dynamic_provisioning.go]: Let's keep it for now

**Pull request merged**